Project/Source/Components/ComponentParticleSystem.cpp

```diff
@@ -0,0 +1,305 @@
+		int width;
+		int height;
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_WIDTH, &width);
+		glGetTextureLevelParameteriv(textureResource->glTexture, 0, GL_TEXTURE_HEIGHT, &height);
+
+		if (oldID != textureID) {
+			ComponentTransform2D* transform2D = GetOwner().GetComponent<ComponentTransform2D>();
+			if (transform2D != nullptr) {
+				transform2D->SetSize(float2(width, height));
+			}
+		}
+
+		ImGui::Text("");
+		ImGui::Separator();
+		ImGui::TextColored(App->editor->titleColor, "Texture Preview");
+		ImGui::TextWrapped("Size:");
+		ImGui::SameLine();
+		ImGui::TextWrapped("%d x %d", width, height);
+		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		ImGui::InputInt("Xtiles: ", &Xtiles);
+		ImGui::InputInt("Ytiles: ", &Ytiles);
+		ImGui::InputFloat("Scale: ", &scale);
+		ImGui::InputFloat("Life: ", &particleLife);
+
+		if (ImGui::InputFloat("Speed: ", &velocity)) {
+			CreateParticles(maxParticles, velocity);
+		}
+
+		if (ImGui::InputFloat("MaxParticles: ", &maxParticles)) {
+			CreateParticles(maxParticles, velocity);
+		}
+
+		ImGui::ColorEdit4("InitColor##", initC.ptr());
+		ImGui::ColorEdit4("FinalColor##", finalC.ptr());
+	}
+}
+
+float3 ComponentParticleSystem::CreateVelocity() {
+	if (emitterType == EmitterType::CONE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float z = (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float y = (float(rand()) / float((RAND_MAX)) * 1.0);
+		return float3(x, y, z);
+	}
+	if (emitterType == EmitterType::SPHERE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		float z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		float y = (transform->GetGlobalPosition().y) + (float(rand()) / float((RAND_MAX)) * 2.0) - 1.0f;
+		return float3(x, y, z);
+	}
+};
+
+float3 ComponentParticleSystem::CreatePosition() {
+	if (emitterType == EmitterType::CONE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 0.2) - 0.2;
+		float y = (transform->GetGlobalPosition().y);
+		return float3(x, y, z);
+	}
+	if (emitterType == EmitterType::SPHERE) {
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float x = (transform->GetGlobalPosition().x) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		float z = (transform->GetGlobalPosition().z) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		float y = (transform->GetGlobalPosition().y) + (float(rand()) / float((RAND_MAX)) * 0.5) - 0.5;
+		return float3(transform->GetGlobalPosition());
```

Comment:
Hardcoded values should be maked with a TODO so that we know what to change later.